### PR TITLE
lib: change event signature

### PR DESCRIPTION
**Pull request by lundibundi**
Changes event message signature to always send an array
of arbitrary arguments. This makes it work same way as
EventEmitter does and also makes its usage similar to
call action.

Also the tests are just plainly awesome, but we will rewrite them soon anyway.

### CODE REVIEW

File: lib/remote-proxy.js [hunk 1]
```diff
@@ -24,22 +24,22 @@ function RemoteProxy(connection, interfaceName, methods = []) {
...
+  this._emitLocal(eventName, eventArgs);
+};
 
-  events.EventEmitter.prototype.emit.call(this, eventName, eventArgs);
+// Emit local event.
+//   eventName - name of an event
+//   eventArgs - array of event arguments
+//
+RemoteProxy.prototype._emitLocal = function(eventName, eventArgs) {
+  if (!eventArgs) eventArgs = [];
```
Comment (belochub) [inline lib/remote-proxy.js]: You should use default parameters instead of this.

**Comment by aqrln** [comment]
@belochub as far as I can tell from the in-person conversation, you weren't extremely happy with this change. Did you change your mind or...?

Defensively marking this as blocked until I'm sure there is consensus about this.

**Comment by belochub** [comment]
@aqrln, yes, it is okay, I approved the PR after @lundibundi explained to me the point of it.

**Comment by lundibundi** [comment]
Landed in 852b46c0ac488eb36a8733f18bc6a2b1fed1f153.

**Pull request closed without merging**